glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -327,34 +331,8 @@ open class GleanInternalAPI internal constructor () {
         branch: String,
         extra: Map<String, String>? = null
     ) {
-        // The Map is sent over FFI as a pair of arrays, one containing the
-        // keys, and the other containing the values.
-        // In Kotlin, Map.keys and Map.values are not guaranteed to return the entries
-        // in any particular order. Therefore, we iterate over the pairs together and
-        // create the keys and values arrays step-by-step.
-        var keys: StringArray? = null
-        var values: StringArray? = null
-        var numKeys = 0
-
-        extra?.let {
-            numKeys = extra.size
-            val extraList = extra.toList()
-            keys = StringArray(Array(extra.size) { extraList[it].first }, "utf-8")
-            values = StringArray(Array(extra.size) { extraList[it].second }, "utf-8")
-        }
-
-        // We dispatch this asynchronously so that, if called before the Glean SDK is
-        // initialized, it doesn't get ignored and will be replayed after init.
-        @Suppress("EXPERIMENTAL_API_USAGE")
-        Dispatchers.API.launch {
-            LibGleanFFI.INSTANCE.glean_set_experiment_active(
-                experimentId,
-                branch,
-                keys,
-                values,
-                numKeys
-            )
-        }
+        var map = extra ?: mapOf()
+        gleanSetExperimentActive(experimentId, branch, map)
```

Comment:
Decided to just take a non-nullable value here and let the SDK handle the "old" API.
We can just as easily take an optional value in the UDL though.